mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/store/JobArtifact.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.server.master.store;
+
+import io.mantisrx.shaded.com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import io.mantisrx.shaded.com.fasterxml.jackson.annotation.JsonProperty;
+import java.time.Instant;
+import java.util.Map;
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class JobArtifact {
+    @JsonProperty("artifactID")
+    ArtifactID artifactID;
+    @JsonProperty("project")
+    String project;
+    @JsonProperty("version")
+    String version;
+    @JsonProperty("createdAt")
+    Instant createdAt;
+    // Potential values: spring-boot, guice
+    @JsonProperty("runtimeType")
+    String runtimeType;
```

Comment:
This feels too tied to Netflix. Did you have a chance to consider my earlier proposal of having generic tags instead?

Reply:
I don't think having a map for any extra config is a good idea. we need much better validation on which key is present, changing clients will be harder if we change the keys in the map etc etc. I think runtime type is also something other companies can leverage in case of migrations. thoughts?